systems/physics/src/lib.rs

```diff
@@ -12,52 +12,117 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+extern crate decscloud_codec as codec;
 extern crate serde_json;
 extern crate wascap_guest as guest;
 
+use codec::systemmgr::System;
 use guest::prelude::*;
 use serde_json::Value;
 use stacktrader_types as trader;
 use trader::components::*;
 
 call_handler!(handle_call);
 
+const NO_MESSAGE: &str = "(no message)";
+const REGISTRY_SUBJECT: &str = "decs.system.registry";
+
 pub fn handle_call(ctx: &CapabilitiesContext, operation: &str, msg: &[u8]) -> CallResult {
     match operation {
-        messaging::OP_DELIVER_MESSAGE => handle_frame(ctx, msg),
+        messaging::OP_DELIVER_MESSAGE => handle_message(ctx, msg),
         core::OP_HEALTH_REQUEST => Ok(vec![]),
         _ => Err("bad dispatch".into()),
     }
 }
 
-/// Receives messages on the subject `systems.{shard}.{system}.frames`, e.g. `systems.default.physics.frames`
-fn handle_frame(
+/// Routes message either to the `handle_ping` function for registry pings or `handle_frame` for position updates
+fn handle_message(
     ctx: &CapabilitiesContext,
     msg: impl Into<messaging::DeliverMessage>,
 ) -> CallResult {
     let msg = msg.into().message;
+    let subject = msg
+        .as_ref()
+        .map_or(NO_MESSAGE.to_string(), |m| m.subject.to_string());
     ctx.log(&format!(
         "Received message from broker on subject '{}'",
-        msg.as_ref()
-            .map_or("(no message)".to_string(), |m| m.subject.to_string())
+        subject
     ));
-    let bodyvec = msg.unwrap().body;
+    match subject.as_ref() {
+        NO_MESSAGE => Err("No message".into()),
+        REGISTRY_SUBJECT => handle_ping(ctx, msg.unwrap()),
+        _ => handle_frame(ctx, msg.unwrap()),
+    }
+}
+
+/// Receives messages on the subject `system.registry` and replies with physics system metadata
+fn handle_ping(
+    ctx: &CapabilitiesContext,
+    msg: guest::prelude::messaging::BrokerMessage,
+) -> CallResult {
+    let payload = System {
+        name: "physics".to_string(),
+        framerate: 60,
+        components: vec!["Position".to_string(), "Velocity".to_string()],
```

Comment:
Components should, by convention, all be lowercase. We can probably ensure that we're lowercasing them everywhere they're referenced, but we should be specific. Might also want to convert these into constants.

Reply:
Refactored this code to use constants with lowercase lettering